### ISSUE TITLE
CUDA Unit test failue for FixedRadiusIndex.SearchRadiusBatch

### ISSUE THREAD

**Issue by yxlao** [isl-org/Open3D#3856]
## Issue

```bash
make tests -j && ./bin/tests --gtest_filter="*FixedRadiusIndex.SearchRadiusBatch*"
```

```txt
Note: Google Test filter = *FixedRadiusIndex.SearchRadiusBatch*
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from FixedRadiusIndex
[ RUN      ] FixedRadiusIndex.SearchRadiusBatch
/home/yixing/repo/Open3D/cpp/tests/test_utility/Compare.cpp:69: Failure
Expected equality of these values:
  v0[i]
    Which is: 9
  v1[i]
    Which is: 6
/home/yixing/repo/Open3D/cpp/tests/test_utility/Compare.cpp:69: Failure
Expected equality of these values:
  v0[i]
    Which is: 6
  v1[i]
    Which is: 9
/home/yixing/repo/Open3D/cpp/tests/test_utility/Compare.cpp:69: Failure
Expected equality of these values:
  v0[i]
    Which is: 8
  v1[i]
    Which is: 6
/home/yixing/repo/Open3D/cpp/tests/test_utility/Compare.cpp:69: Failure
Expected equality of these values:
  v0[i]
    Which is: 6
  v1[i]
    Which is: 8
/home/yixing/repo/Open3D/cpp/tests/test_utility/Compare.cpp:69: Failure
Expected equality of these values:
  v0[i]
    Which is: 18
  v1[i]
    Which is: 16
/home/yixing/repo/Open3D/cpp/tests/test_utility/Compare.cpp:69: Failure
Expected equality of these values:
  v0[i]
    Which is: 16
  v1[i]
    Which is: 18
/home/yixing/repo/Open3D/cpp/tests/test_utility/Compare.cpp:69: Failure
Expected equality of these values:
  v0[i]
    Which is: 19
  v1[i]
    Which is: 16
/home/yixing/repo/Open3D/cpp/tests/test_utility/Compare.cpp:69: Failure
Expected equality of these values:
  v0[i]
    Which is: 16
  v1[i]
    Which is: 19
[  FAILED  ] FixedRadiusIndex.SearchRadiusBatch (317 ms)
[----------] 1 test from FixedRadiusIndex (317 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (317 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] FixedRadiusIndex.SearchRadiusBatch

```

## Environment
- Operating system: Ubuntu 20.04
- Open3D version: `master`  76c2baf9debd460900f056a9b51e9a80de9c0e64
- CUDA 11.0
- GTX 1080Ti

**Comment by reyanshsolis** [comment]
Same result on my system [CUDA 11.0, Ubuntu 20.04, GTX 1050Ti, i7-7700]
```
[----------] 4 tests from FixedRadiusIndex
[ RUN      ] FixedRadiusIndex.SearchRadius
[       OK ] FixedRadiusIndex.SearchRadius (0 ms)
[ RUN      ] FixedRadiusIndex.SearchRadiusBatch
/home/rey/workspace/isl/treg/Open3D/cpp/tests/test_utility/Compare.cpp:69: Failure
Expected equality of these values:
  v0[i]
    Which is: 9
  v1[i]
    Which is: 6
/home/rey/workspace/isl/treg/Open3D/cpp/tests/test_utility/Compare.cpp:69: Failure
Expected equality of these values:
  v0[i]
    Which is: 6
  v1[i]
    Which is: 9
/home/rey/workspace/isl/treg/Open3D/cpp/tests/test_utility/Compare.cpp:69: Failure
Expected equality of these values:
  v0[i]
    Which is: 8
  v1[i]
    Which is: 6
/home/rey/workspace/isl/treg/Open3D/cpp/tests/test_utility/Compare.cpp:69: Failure
Expected equality of these values:
  v0[i]
    Which is: 6
  v1[i]
    Which is: 8
/home/rey/workspace/isl/treg/Open3D/cpp/tests/test_utility/Compare.cpp:69: Failure
Expected equality of these values:
  v0[i]
    Which is: 18
  v1[i]
    Which is: 16
/home/rey/workspace/isl/treg/Open3D/cpp/tests/test_utility/Compare.cpp:69: Failure
Expected equality of these values:
  v0[i]
    Which is: 16
  v1[i]
    Which is: 18
/home/rey/workspace/isl/treg/Open3D/cpp/tests/test_utility/Compare.cpp:69: Failure
Expected equality of these values:
  v0[i]
    Which is: 19
  v1[i]
    Which is: 16
/home/rey/workspace/isl/treg/Open3D/cpp/tests/test_utility/Compare.cpp:69: Failure
Expected equality of these values:
  v0[i]
    Which is: 16
  v1[i]
    Which is: 19
[  FAILED  ] FixedRadiusIndex.SearchRadiusBatch (0 ms)
```

**Comment by junha-l** [comment]
I couldn't reproduce the error in my environment
```
(open3d) root@postech:~/code/Open3D/build_cuda11# ./bin/tests ^C
(open3d) root@postech:~/code/Open3D/build_cuda11# nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2020 NVIDIA Corporation
Built on Wed_Jul_22_19:09:09_PDT_2020
Cuda compilation tools, release 11.0, V11.0.221
Build cuda_11.0_bu.TC445_37.28845127_0
(open3d) root@postech:~/code/Open3D/build_cuda11# cat /etc/os-release
NAME="Ubuntu"
VERSION="18.04.5 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.5 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
(open3d) root@postech:~/code/Open3D/build_cuda11# python --version
Python 3.8.11
(open3d) root@postech:~/code/Open3D/build_cuda11# ./bin/tests --gtest_filter="*FixedRadiusIndex*"
[Open3D INFO] CPUInfo: 8 cores, 8 threads.
Note: Google Test filter = *FixedRadiusIndex*
[==========] Running 4 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 4 tests from FixedRadiusIndex
[ RUN      ] FixedRadiusIndex.SearchRadius
[       OK ] FixedRadiusIndex.SearchRadius (393 ms)
[ RUN      ] FixedRadiusIndex.SearchRadiusBatch
[       OK ] FixedRadiusIndex.SearchRadiusBatch (0 ms)
[ RUN      ] FixedRadiusIndex.SearchHybrid
[       OK ] FixedRadiusIndex.SearchHybrid (0 ms)
[ RUN      ] FixedRadiusIndex.SearchHybridBatch
[       OK ] FixedRadiusIndex.SearchHybridBatch (0 ms)
[----------] 4 tests from FixedRadiusIndex (394 ms total)

[----------] Global test environment tear-down
[==========] 4 tests from 1 test suite ran. (394 ms total)
[  PASSED  ] 4 tests.
```

By the way, I guess the failure is in line 117 of ```cpp/tests/core/FixedRadiusIndex.cpp```, where it checks the unsorted indices. I made the ground-truth unsorted indices using pytorch wrapper of FixedRadiusSearch, and it was compiled on CUDA 10.1. Somehow the order of indices is different but I think it is not a big problem since it is ```unsorted`` indices anyway. 

To resolve the CI error, we can disable the checks for unsorted indices.

**Comment by junha-l** [comment]
Resolved #3895
